DatadogFlags/Sources/FlagsStore.swift

```diff
@@ -5,12 +5,23 @@
  */
 
 import Foundation
+import DatadogInternal
 
 // TODO: FFL-1048 Use the Core SDK’s DataStore instead
 internal class FlagsStore {
-    private var cachedFlags: [String: Any] = [:]
-    private var flagsMetadata: FlagsMetadata?
-    private let syncQueue = DispatchQueue(label: "com.datadoghq.flags.store", attributes: .concurrent)
+    private struct State: Codable {
```

Comment:
Ah, so we would store everything in the same `State`. That's neater than storing flag assignments and metadata separately 👍

Reply:
Exactly. On top of that, it avoids impossible states, like non-empty flag assignments with a `nil` evaluation context.